samcli/lib/telemetry/event.py

```diff
@@ -39,6 +41,23 @@ class EventType:
             "CreateChangeSetSuccess",
         ],
         EventName.BUILD_FUNCTION_RUNTIME: INIT_RUNTIMES,
+        EventName.SYNC_USED: [
+            "Start",
+            "End",
+        ],
+        EventName.SYNC_FLOW: [
```

Comment:
Why don't we just use the class name of the classes defined under "samcli/lib/sync/flows/..."? 
We can probably update samcli/lib/sync/flows/__init__.py to first collect all SyncFlow classes and sub-classes, and then just do something like `from samcli.lib.sync.flows import *` to import all these classes/sub-classes. If we do this change, we might also want to update other places where these SyncFlow classes/sub-classes are imported.

Reason for asking for this change is mainly to reduce our manual effort needed to maintain this list in the future

Reply:
I considered looking into this, but there wasn't a consistent list I could find to collect all the SyncFlows, especially since some files included additional resources (i.e. abstract classes, module functions). If we don't anticipate many additional SyncFlows to be added, I don't suspect it should be much more of a hassle to add them into the tracker.

Reply:
Another thought - is it possible to do the event registration and event tracking in the `__init__` method of a SyncFlow subclass?

For example, within the `__init__` method of `AliasVersionSyncFlow`:
1. append  `AliasVersionSyncFlow.__name__` (or `type(self).__name__`) to the list `EventName.SYNC_FLOW` if it is not in it already
2. `EventTracker.track_event("SyncFlow", type(self).__name__)`

We might just create a method to do both in the `SyncFlow` class, then every subclass of `SyncFlow` would have it called automatically.

Just trying to think of ways that we don't need to do anything manually.

Reply:
It's possible, but my worry is that creating a method to add new values dynamically could potentially lead to misuse (i.e. one wrong line accidentally storing a possible value as some arbitrary string, or Events not being stored when they should be). The hard-coded route does create more of a hassle for the implementation, but unless the `SyncFlow` classes themselves (or other modules that implement them) are refactored in such a way that we have a concrete list of all SyncFlows, I think it's still the safest solution.

Reply:
okay it makes sense to me about your concern in the event registration part